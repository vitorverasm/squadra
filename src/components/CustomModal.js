import React from 'react';
import {
  Modal, StyleSheet, View, TouchableOpacity
} from 'react-native';
import Icon from 'react-native-vector-icons/Feather';
import { colors } from '../utils/styles';

const CustomModal = ({
  visible, close, content, height
}) => (
  <Modal visible={visible} transparent animationType="slide" onRequestClose={close}>
    <View style={styles.outside}>
      <View style={[styles.modal, { height: height || '50%' }]}>
        <View style={styles.iconContainer}>
          <TouchableOpacity onPress={close}>
            <Icon
              name="x"
              size={25}
              color={colors.primary}
              style={{ backgroundColor: 'transparent' }}
            />
          </TouchableOpacity>
        </View>
        <View style={styles.childrenContainer}>{content || null}</View>
      </View>
    </View>
  </Modal>
);

const styles = StyleSheet.create({
  childrenContainer: {
    flex: 1,
    marginHorizontal: 24,
    marginTop: 59,
    marginBottom: 43
  },
  iconContainer: {
    position: 'absolute',
    top: 26,
    left: 24,
    justifyContent: 'center',
    alignItems: 'center'
  },
  modal: {
    height: '50%',
    width: '100%',
    backgroundColor: colors.white,
    borderTopLeftRadius: 10,
    borderTopRightRadius: 10
  },
  outside: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'flex-end',
    backgroundColor: colors.modalOpacity
  }
});

export default CustomModal;
