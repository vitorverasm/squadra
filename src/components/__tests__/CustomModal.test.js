import { shallow } from 'enzyme';
import React from 'react';
import { View } from 'react-native';
import CustomModal from '../CustomModal';

describe('CustomModal', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<CustomModal />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders with props', () => {
    const wrapper = shallow(
      <CustomModal visible close={jest.fn} content={<View />} height="50%" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
