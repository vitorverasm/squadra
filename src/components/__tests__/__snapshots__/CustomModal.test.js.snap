// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomModal renders correctly 1`] = `
"<Component visible={true} transparent={true} animationType=\\"slide\\" onRequestClose={[undefined]} hardwareAccelerated={false}>
  <View style={{...}}>
    <View style={{...}}>
      <View style={{...}}>
        <TouchableOpacity onPress={[undefined]} activeOpacity={0.2}>
          <Icon name=\\"x\\" size={25} color=\\"#FF3B30\\" style={{...}} allowFontScaling={false} />
        </TouchableOpacity>
      </View>
      <View style={{...}} />
    </View>
  </View>
</Component>"
`;

exports[`CustomModal renders with props 1`] = `
"<Component visible={true} transparent={true} animationType=\\"slide\\" onRequestClose={[Function: bound fn]} hardwareAccelerated={false}>
  <View style={{...}}>
    <View style={{...}}>
      <View style={{...}}>
        <TouchableOpacity onPress={[Function: bound fn]} activeOpacity={0.2}>
          <Icon name=\\"x\\" size={25} color=\\"#FF3B30\\" style={{...}} allowFontScaling={false} />
        </TouchableOpacity>
      </View>
      <View style={{...}}>
        <View />
      </View>
    </View>
  </View>
</Component>"
`;
